### PR TITLE
Clarifai-LiteLLM Addressed proxy issues

### LINKED ISSUE #3862: [Bug]: clarifai provider in litellm[proxy] not work
### What happened?

My config.yaml:
- model_name: clarifai-gpt-4o
    litellm_params:
      model: clarifai/openai.chat-completion.gpt-4o
      api_base: https://clarifai.com/v1/
      api_key: CLARIFAI_API_KEY
      rpm: 100

Athought python code worked well but liteLLM proxy did not work successfully.
import os
from litellm import completion

os.environ["CLARIFAI_API_KEY"] = MY_KEY

response = completion(
  model="clarifai/openai.chat-completion.openai-gpt-4o",
  messages=[{ "content": "Who you are ?","role": "user"}]
)

print(response)

So pls help me fix this issue.

### Relevant log output

```shell
HTTP 500 Internal Server Error from http://127.0.0.1:4000/v1/chat/completions {"error":{"message":"sequence item 0: expected str instance, list found\nNumber Retries = 1","type":null,"param":null,"code":500}} Code: undefined Error number: undefined Syscall: undefined Type: undefined Error: HTTP 500 Internal Server Error from http://127.0.0.1:4000/v1/chat/completions {"error":{"message":"sequence item 0: expected str instance, list found\nNumber Retries = 1","type":null,"param":null,"code":500}} at customFetch
```

### Twitter / LinkedIn details

_No response_

### PR DESCRIPTION
**Pull request by mogith-pn**
## Relevant issues

* Fixes #3862 & #3769

## Type

🐛 Bug Fix
📖 Documentation

## Changes

* Modified Async completion function.
* Modified stream function.
* Addressed spelling mistakes in doc.

## [REQUIRED] Testing - Attach a screenshot of any new tests passing locall
If UI changes, send a screenshot/GIF of working UI fixes
Screenshot 2024-06-10 at 11 59 41 PM

### CODE REVIEW

**Comment by mogith-pn** [comment]
@ishaan-jaff  

Can you please take a look into this PR ? Thanks !

File: litellm/llms/clarifai.py [hunk 1]
```diff
@@ -245,7 +252,7 @@ def completion(
         },
     )
     if acompletion == True:
-        return async_completion(
+        return asyncio.run(async_completion(
```
Comment (ghost) [inline litellm/llms/clarifai.py:255]: @mogith-pn can we please have this be actually async. 

Current approach will be a blocking call for users under high traffic.

File: litellm/utils.py [hunk 2]
```diff
@@ -11046,7 +11046,7 @@ def handle_clarifai_completion_chunk(self, chunk):
             completion_tokens = len(encoding.encode(text))
             return {
                 "text": text,
-                "is_finished": True,
+                "is_finished": False,
```
Comment (ghost) [inline litellm/utils.py:11049]: please add testing for this change

**Review by ghost — CHANGES_REQUESTED** [review]
Please make the calls actually async and add testing for async completion + async streaming.

**Comment by ishaan-jaff** [comment]
closing in favor of 
https://github.com/BerriAI/litellm/pull/4158

**Pull request closed without merging**